x-pack/plugins/ml/public/application/explorer/alerts/get_alerts_summary.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ALERT_DURATION,
+  ALERT_RULE_NAME,
+  ALERT_STATUS,
+  ALERT_END,
+  ALERT_START,
+} from '@kbn/rule-data-utils';
+import { groupBy } from 'lodash';
+import { AnomalyDetectionAlert } from './anomaly_detection_alerts_state_service';
+
+export type RulesSummary = Array<[string, RuleSummary]>;
+
+export interface RuleSummary {
+  activeCount: number;
+  totalCount: number;
+  lastDuration: number;
+  startedAt: number;
+  recoveredAt: number | undefined;
+}
+
+export function getAlertsSummary(alertsData: AnomalyDetectionAlert[]): RulesSummary {
+  return Object.entries(groupBy(alertsData, ALERT_RULE_NAME) ?? [])
+    .map<[string, RuleSummary]>(([ruleName, alerts]) => {
+      // Find the latest alert for each rule
+      const latestAlert: AnomalyDetectionAlert = alerts.reduce((latest, alert) => {
+        return alert[ALERT_START] > latest[ALERT_START] ? alert : latest;
+      });
+
+      return [
+        ruleName,
+        {
+          totalCount: alerts.length,
+          activeCount: alerts.filter((alert) => alert[ALERT_STATUS] === 'active').length,
```

Comment:
super nit: Since 'active' is referenced throughout, can be it be a constant?

Reply:
Refactored in 740fe6bdaf906d1ed078c8dc769cfb22bb513a5a